### PR TITLE
Save new credentials

### LINKED ISSUE #96: [Feature:Return] Save credentials when logging in
Prompt to save credentials when a site is logged in to.

### PR DESCRIPTION
**Pull request by perry-mitchell**
Closes #96

### CODE REVIEW

File: source/setup/components/SaveCredentialsPage.js [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+    height: 300px;
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+`;
+const NoArchivesNotice = styled.span`
+    font-size: 13px;
+    color: rgb(242, 159, 4);
+`;
```
Comment (sallar) [inline source/setup/components/SaveCredentialsPage.js]: Would be nice to have an empty line here

File: source/tab/LoginTracker.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { getCurrentTitle, getCurrentURL } from "./page.js";
+
+let __sharedTracker = null;
+
+export default class LoginTracker {
+    constructor() {
+        this.username = "";
```
Comment (sallar) [inline source/tab/LoginTracker.js]: you can just class properties here (similar to `state = ...`) etc.

File: source/tab/index.js [hunk 3]
```diff
@@ -1,21 +1,48 @@
...
+                    },
+                    () => {
+                        const tracker = getSharedTracker();
+                        const now = new Date();
+                        transferLoginCredentials({
+                            username: tracker.username,
+                            password: tracker.password,
+                            url: tracker.url,
+                            title: tracker.title,
+                            timestamp: now.getTime()
```
Comment (sallar) [inline source/tab/index.js]: You can use `Date.now()` here

File: source/setup/components/SaveCredentialsPage.js (hunk 1 above)
Comment (perry-mitchell) [reply source/setup/components/SaveCredentialsPage.js]: I usually group constants by their type/context.. It should be the same in the other files. Simple values, styled components, required modules etc..

File: source/tab/LoginTracker.js (hunk 2 above)
Comment (perry-mitchell) [reply source/tab/LoginTracker.js]: 👍

File: source/setup/components/SaveCredentialsPage.js (hunk 1 above)
Comment (sallar) [reply source/setup/components/SaveCredentialsPage.js]: ah it makes sense then

Comment (perry-mitchell) [reply source/setup/components/SaveCredentialsPage.js]: ☺️

**Pull request merged**